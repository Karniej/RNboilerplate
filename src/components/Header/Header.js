import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native'
import styles from './styles'

const Header = ({ title, additionalStyles }) => {
  return <Text
    style={[styles.header, additionalStyles]}
  >{title}
  </Text>
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
  additionalStyles: PropTypes.object
}
export default Header
