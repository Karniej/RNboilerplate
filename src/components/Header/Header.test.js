import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Header from './Header'

configure({ adapter: new Adapter() })

test('Renders correctly, expect onpress to be defined, but not called', () => {
  const defaultProps = {
    title: 'Header'
  }
  const wrapper = shallow(<Header
    {...defaultProps}
  />)
  expect(wrapper).toMatchSnapshot()
})
