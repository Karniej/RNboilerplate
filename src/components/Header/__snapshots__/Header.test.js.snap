// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly, expect onpress to be defined, but not called 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="Header"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "accessible": true,
      "allowFontScaling": true,
      "children": "Header",
      "ellipsizeMode": "tail",
      "style": Array [
        Object {
          "fontSize": 20,
          "margin": 10,
          "textAlign": "center",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": "Header",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Header",
        "ellipsizeMode": "tail",
        "style": Array [
          Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": "Header",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
